### PR TITLE
Update pywin32 to version 305

### PR DESCRIPTION
**Pull request by AndrewVallette**
## Changes
- Updated version to 305
- Removed numerous stale patches and headers
- Added patches for `setup.py` so that the MSVC folders can be found in our build system
- Added new test to verify that a Windows GUI app can be created

## Links
Github: https://github.com/mhammond/pywin32/
Changelog: https://github.com/mhammond/pywin32/blob/main/CHANGES.txt

## Notes
- The `setup.py` post-install script has been causing problems for builders of this package for at least 5 years. The changes in this PR should prevent the issue from occurring again.

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
The package is currently failing in some architectures, it might be a good idea to see if we can get the package to build using concourse, if not, then we might have to specify on the comments the architectures that are not supported and we might have to skip them for now. 

Regarding `pywin32` in many cases we have to do a manual build in the windows architectures due to some build messages that require manual inputs before the build can complete.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,43 +1,56 @@
...
+    - python run_test.py
+    - python test_win32gui.py
+
 about:
   home: https://github.com/mhammond/pywin32
   # For more detail on the license look at the following issue
   # https://github.com/mhammond/pywin32/issues/1127
-  license: HPND AND BSD-3-Clause AND LGPL-2.0
+  license: HPND AND BSD-3-Clause AND LGPL-2.0-or-later
   license_file: Pythonwin/License.txt
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:54]: The upstream source has multiple license files:
```
 license_file:
    - Pythonwin/License.txt  # BSD-3-CLause
    - Pythonwin/Scintilla/License.txt  # MIT
    - win32/License.txt  # BSD-3-CLause
    - com/License.txt  # BSD-3-CLause
    - com/win32comext/mapi/src/mapi_stub_library/LICENSE  # MIT
    - adodbapi/license.txt  # LGPL-2.1-or-later
```
I didn't find mentions of the `HPND` license

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,43 +1,56 @@
...
 
+test:
+  files:
+    - test_win32gui.py
+  requires:
+    - pip
+    - python
+  commands:
+    - pip check
+    - python run_test.py
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:46]: An error on `win64`:
```
  File "C:\b\abs_627wkvj_i7\croot\pywin32_1669249312158\test_tmp\run_test.py", line 53, in <module>

    assert os.path.isfile(pythoncom_filename)

AssertionError
```

**Comment by skupr-anaconda** [comment]
I don't know why but conda-forge  doesn't build scintilla https://github.com/conda-forge/pywin32-feedstock/blob/main/recipe/do-not-build-scintilla.patch

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:46]: If you add to run_test.py:
```
import glob

print(
    list(
        glob.glob(
            os.path.join(
                os.environ["LIBRARY_BIN"],
                'py*.dll'
            )
        )
    )
)
```
the result will be the empty list `[]`

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: There is one win32com demo *.py file using HPND license.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,43 +1,56 @@
...
     - setuptools
   run:
     - python
 
+test:
+  files:
+    - test_win32gui.py
+  requires:
+    - pip
+    - python
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml]: done

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,43 +1,56 @@
...
     - {{ compiler("c") }}
-    - m2-patch  # [win]
-    - m2-gcc-libs  # [win]
-    - patch  # [not win]
+    - m2-patch
+    - m2-gcc-libs
   host:
     - python
     - pip
     - setuptools
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml:35]: done

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml:47]: The latest added patch solves this issue.

**Comment by AndrewVallette** [comment]
> I don't know why but conda-forge doesn't build scintilla https://github.com/conda-forge/pywin32-feedstock/blob/main/recipe/do-not-build-scintilla.patch

They seemed to have some MSVC path issues, and disabled it as a workaround. Here, I have included a patch containing some `setup.py` changes from upstream that they don't have yet, and this handles MSVC paths better.

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by katietz — APPROVED** [review]
Thanks.  Just one nit I saw.  We should change .gitattributes file so that patch/diff files use 'text eol=lf' instead of binary.  As we use a tar-file, the line-endings should be always the same

**Pull request merged**